src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java

```diff
@@ -548,80 +615,88 @@ private void setTemplateAttributes(String jsonString, Map<String, Object> attrib
-					} else {
-						attribute.put(value, String.valueOf(object));
 					}
 				}
-					
-				}
 			}
-
 		} catch (JsonParseException | JsonMappingException e) {
 			printLogger.error("Error while parsing Json file" ,e);
 			throw new ParsingException(PlatformErrorMessages.PRT_RGS_JSON_PARSING_EXCEPTION.getMessage(), e);
 		}
 	}
 
-	/**
-	 * Gets the password.
-	 *
-	 * @param uin
-	 *            the uin
-	 * @return the password
-	 * @throws IdRepoAppException
-	 *             the id repo app exception
-	 * @throws NumberFormatException
-	 *             the number format exception
-	 * @throws ApisResourceAccessException
-	 *             the apis resource access exception
-	 * @throws IOException
-	 *             Signals that an I/O exception has occurred.
-	 */
-	private String getPassword(String uin) throws ApisResourceAccessException, IOException {
-		JSONObject jsonObject = utilities.retrieveIdrepoJson(uin);
-
-		String[] attributes = env.getProperty(UINCARDPASSWORD).split("\\|");
-		List<String> list = new ArrayList<>(Arrays.asList(attributes));
-
-		Iterator<String> it = list.iterator();
-		String uinCardPd = "";
-
-		while (it.hasNext()) {
-			String key = it.next().trim();
-
-			Object object = JsonUtil.getJSONValue(jsonObject, key);
-			if (object instanceof ArrayList) {
-				JSONArray node = JsonUtil.getJSONArray(jsonObject, key);
-				JsonValue[] jsonValues = JsonUtil.mapJsonNodeToJavaObject(JsonValue.class, node);
-				uinCardPd = uinCardPd.concat(getParameter(jsonValues, templateLang));
-
-			} else if (object instanceof LinkedHashMap) {
-				JSONObject json = JsonUtil.getJSONObject(jsonObject, key);
-				uinCardPd = uinCardPd.concat((String) json.get(VALUE));
-			} else {
-				uinCardPd = uinCardPd.concat((String) object);
-			}
-
-		}
-
-		return uinCardPd;
-	}
+    /**
+     * Gets the password.
+     *
+     * @param jsonObject
+     * @return
+     * @throws Exception
+     */
+    private String getPassword(org.json.JSONObject jsonObject) throws ApisResourceAccessException, IOException {
+
+        String[] attributes = env.getProperty(UINCARDPASSWORD).split("\\|");
+        List<String> list = new ArrayList<>(Arrays.asList(attributes));
+        Iterator<String> it = list.iterator();
+        String uinCardPd = "";
+        Object obj = null;
+        while (it.hasNext()) {
+            String key = it.next().trim();
+            Object object = jsonObject.get(key);
+            if (object != null) {
+                try {
+                    obj = new JSONParser().parse(object.toString());
+                } catch (Exception e) {
+                    obj = object;
+                }
+            }
+            if (obj instanceof JSONArray) {
+                JsonValue[] jsonValues = JsonUtil.mapJsonNodeToJavaObject(JsonValue.class, (JSONArray) obj);
+                uinCardPd = uinCardPd.concat(getFormattedPasswordAttribute(getParameter(jsonValues, templateLang)).substring(0, 4));
```

Comment:
any reason we are hardcoding as 4 here ?

Reply:
Also why this is not followed for object else if case ?

Reply:
taken care.